### PR TITLE
Converted data frame addressing examples to a challenge.

### PR DESCRIPTION
**Pull request by tomwright01**
I found working through the data frame addressing examples in 04-data-structures-part-1 to be very valuable. I think it is worth promoting this to a challenge.

Writing the solution was not as easy as I anticipated. If anyone has suggestions for clarification I'm happy to hear them.

### CODE REVIEW

File: 04-data-structures-part1.Rmd [hunk 1]
```diff
@@ -486,8 +490,40 @@ ncol(matrix_example)
...
+> Again we use the single brace with row and column coordinates. The column coordinate is not specified. The return value is a _list_ containing all the values in the first row.
 >
-> ## Solution to challenge 3 {.challenge}
+
+
+>
+> ## Solution to challenge 4 {.challenge}
 >
 > What do you think will be the result of
 > `length(matrtix_example)`?
```
Comment (naupaka) [inline 04-data-structures-part1.Rmd]: `matrtix_example` should be `matrix_example`

File: 04-data-structures-part1.Rmd [hunk 2]
```diff
@@ -486,8 +490,40 @@ ncol(matrix_example)
...
+> ```{r, eval=TRUE, echo=TRUE}
+> cats$coat
+> ```
+> This example uses the `$` character to address items by name. _coat_ is the first column of the data frame, again a _vector_ of type _factor_.
+> ```{r, eval=TRUE, echo=TRUE}
+> cats["coat"]
+> ```
+> Here we are using a single brace `["coat"]` replacing the index number with the column name. Like example 1, the returned object is a _list_.
+> ```{r, eval=TRUE, echo=TRUE}
+> cats[1,1]
```
Comment (naupaka) [inline 04-data-structures-part1.Rmd]: I think the Google R style guide (which we generally try to follow) recommends a space after a comma. E.g. `cats[1,1]` to `cats[1, 1]` and also in the few lines below (515 and 519) as well.

File: 04-data-structures-part1.Rmd [hunk 3]
```diff
@@ -486,8 +490,40 @@ ncol(matrix_example)
...
+> ```{r, eval=TRUE, echo=TRUE}
+> cats[1,]
+> ```
+> Again we use the single brace with row and column coordinates. The column coordinate is not specified. The return value is a _list_ containing all the values in the first row.
 >
-> ## Solution to challenge 3 {.challenge}
+
+
+>
+> ## Solution to challenge 4 {.challenge}
```
Comment (naupaka) [inline 04-data-structures-part1.Rmd]: Capitalize 'challenge'

File: 04-data-structures-part1.Rmd [hunk 4]
```diff
@@ -502,7 +538,7 @@ ncol(matrix_example)
 >
 
 >
-> ## Solution to challenge 4 {.challenge}
+> ## Solution to challenge 5 {.challenge}
```
Comment (naupaka) [inline 04-data-structures-part1.Rmd]: Capitalize 'challenge'

File: 04-data-structures-part1.Rmd [hunk 5]
```diff
@@ -530,7 +566,7 @@ ncol(matrix_example)
 >
 
 >
-> ## Solution to challenge 6 {.challenge}
+> ## Solution to challenge 7 {.challenge}
```
Comment (naupaka) [inline 04-data-structures-part1.Rmd]: Capitalize 'challenge'

**Comment by naupaka** [comment]
Looks good! Nice solutions. I made a couple proofreading suggestions but once those are done looks ready to merge.

**Comment by tomwright01** [comment]
Fixes made, thanks for proofreading.

**Comment by naupaka** [comment]
Great, thanks!

**Pull request merged**